src/HIE/Bios/Cradle.hs

```diff
@@ -487,11 +497,11 @@ stackAction work_dir mc l _fp = do
   -- Same wrapper works as with cabal
   withCabalWrapperTool ghcProcArgs work_dir $ \wrapper_fp -> do
     (ex1, _stdo, stde, args) <-
-      readProcessWithOutputFile l work_dir
-              "stack" $ ["repl", "--no-nix-pure", "--with-ghc", wrapper_fp]
-                        ++ [ comp | Just comp <- [mc] ]
+      readProcessWithOutputFile l $ (proc "stack" (["repl", "--no-nix-pure", "--with-ghc", wrapper_fp]
```

Comment:
Can you add back the directory argument to `readProcessWithOutputFile` and set it inside this function rather than at all the call sites?

Reply:
Agreed, good point.